### PR TITLE
allow hard-deleting cases that are already soft-deleted

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?188216

@snopoke - full disclosure, i haven't actually run this so worth making sure it looks right. cc @orangejenny

### CODE REVIEW

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 1]
```diff
@@ -78,8 +79,8 @@ def safe_hard_delete(case):
     """
     if not settings.UNIT_TESTING:
         from corehq.apps.commtrack.const import USER_LOCATION_OWNER_MAP_TYPE
-        if case.type != USER_LOCATION_OWNER_MAP_TYPE:
-            raise CommCareCaseError("Attempt to hard delete a case whose type isn't white listed")
+        if not (is_deleted(case) or case.type == USER_LOCATION_OWNER_MAP_TYPE):
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: rather do `case.is_deleted`

File: corehq/ex-submodules/casexml/apps/case/cleanup.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 from __future__ import absolute_import
 from xml.etree import ElementTree
+from dimagi.utils.couch.undo import is_deleted
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: F401 'is_deleted' imported but unused

File: corehq/ex-submodules/casexml/apps/case/cleanup.py (hunk 1 above)
Comment (czue) [inline corehq/ex-submodules/casexml/apps/case/cleanup.py]: right of course. updated

**Comment by snopoke** [comment]
:+1:

**Pull request merged**